### ISSUE TITLE
Check constants passed to arguments of `fromJSON` calls are valid as JSON

### ISSUE THREAD

**Issue by rhysd** [rhysd/actionlint#464]
I noticed that it is [popular](https://github.com/search?q=fromJSON%28%27+lang%3Ayaml&type=code) to pass a string constant to `fromJSON()` to create an immediate array/object value. Here is a simple example:

```yaml
on: push

jobs:
  test:
    runs-on: ubuntu-latest
    if: ${{ contains(fromJSON('["refs/heads/main", "refs/heads/master"]'), github.ref) }}
    steps:
      - run: echo hello
```

It would be nice to check the string literals are valid as JSON.

**Comment by rhysd** [comment]
It is also possible to strongly type the return value of `fromJSON` but I'm not sure this is really needed.
